### PR TITLE
Expose comment referer

### PR DESCRIPTION
**Pull request by Igrom**
If a comment referer is a valid URL, protocol and all (excluding localhost), then it is displayed as a link in the comment header.

The hostname (i.e., host excluding port) is used as the label. 

screenshot from 2018-10-11 11-18-23

### CODE REVIEW

File: src/components/Comment.jsx [hunk 1]
```diff
@@ -209,6 +220,7 @@ class Comment extends React.Component {
                     formatter={reactTimeAgoFormatters[this.props.locale]}/></span>
                 : null}
             {icon}
+            {commentRefererElement}
```
[the remark this answers is not in the document]
Comment (Igrom) [reply src/components/Comment.jsx]: "Provide information on where the comment was created" is the name of the issue. Are you against doing it as a principle, or did you envision another way of doing it?

Only URL referrers would be shown, and in a standarized way as well.

[the remark this answers is not in the document]
Comment (Igrom) [reply src/components/Comment.jsx]: Going with the assumption that displaying the referrer is not an inherently bad experience, we could ensure a good experience by:

a) excluding localhost from the collection of shown referrers
b) allowing the user to view the entire URL before following the link
c) communicating when the URL is not a Cimpress domain (e.g., "are you sure you want to go to that location?")
d) checking if the domain exists (e..g,through https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/dns/resolve)

One could argue that a bad experience could be had if e.g., a referer was forged with a swear word. But if a website exists with such a name, it is not fair to hide it away.

[the remark this answers is not in the document]
Comment (istanishev) [reply src/components/Comment.jsx]: I'd propose we simplify a bit. What about having a (possibly) clickable "globe" icon + tooltip that shows "This comment was posted at {{hostname}}. Click to follow."

I think we should limit the use cases by focusing on "Cimpress" domains. Although it is completely possible that someone puts Comments component in a non-cimpress domain, I feel we are not there yet. Let's not show anything for non-cimpress.io/com domains.

If the referrer is a cimpress domain:
a) the domain name is the same as the one the UI has been - lets find an icon that resembles "here" or "home" and just show an icon and tooltip='Comment has been place on this UI"
b) the domain name is different - icon (globe) + link + tooltip ("Comment has been at {{hostname}}. Click to follow".

[the remark this answers is not in the document]
Comment (istanishev) [reply src/components/Comment.jsx]: image
^ this is how much space we have so adding anything else than an icon might be problematic

File: src/components/Comment.jsx [hunk 2]
```diff
@@ -194,6 +196,15 @@ class Comment extends React.Component {
             </div>
         );
 
+        let refererLabel = refererUtils.makeIntoLabel(this.state.commentObject.referer);
```
[the remark this answers is not in the document]
Comment (Igrom) [reply src/components/Comment.jsx]: It just so coincides that the value of the label is the referrer's hostname. If we decide to change the scheme, or show more than just URL referrer, I wouldn't want to have to change the name --- thus I generalized it.

File: src/components/CommentRefererIcon.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                Click the icon to follow the link.
+            </span>);
+
+        let icon = (
+            <div className={`comment-header-icon fa fa-${iconClass}`}/>
+        );
+
+        return (
+            <a href={referer} target="_blank" rel="noopener noreferrer">
+                <Tooltip
```
Comment (istanishev) [inline src/components/CommentRefererIcon.jsx]: Shouldn't this be the other way: <Tooltip><a></a></Tooltip>?

**Comment by rnowosielski** [comment]
I like the direction in which it is going. 

Although the link seems unclickable now (in certain situations)
oct-12-2018 21-39-12

**Comment by istanishev** [comment]
> I like the direction in which it is going.
> 
> Although the link seems unclickable now (in certain situations)
> oct-12-2018 21-39-12
> 
> In the end I didn't manage to follow the link.

I feel that reversing the <a> and <Tooltip> nesting should help with this.

**Comment by rnowosielski** [comment]
I tried that locally and it didn’t seem to matter. What did help, was removing the minwidth, but then the tooltip looked really bad.

I am not sure why this tooltip shows where it shows instead of “top” how it is configred

**Comment by istanishev** [comment]
> em is not an absolute unit - it is a unit that is relative to the currently chosen font size. Unless you have overridden font style by setting your font size with an absolute unit (such as px or pt), this will be affected by the choice of fonts in the user's browser or OS if they have made one, so it does not make sense to use em as a general unit of length except where you specifically want it to scale as the font size scales.

Nice, we shouldn't really use 'em' according to ^^^

(btw, we should still change the nesting if "a" and "Tooltip")

`tooltipStyle` controls the width of the 'tooltip' element but there is another child element with class 'tooltip-inner' which has 200px max-width. This basically restricts the width to 200px which is not enough to show 'templatedesigner.staging.trdelnik.cimpress.io' ;)
image
^ this is with tooltipStyle set to 300px

**Comment by istanishev** [comment]
Ah, also we should update the version of react components as it is v3 which is way to old and latest versions have improvements for tooltips.

**Pull request merged**